### DISCUSSION TITLE
[v5] Validator to Model, snake_case to camelCase

### DISCUSSION THREAD

**Discussion by bitkidd** [adonisjs/core#2204] [Help] [answered]
Hey!
I was trying to figure it out for myself what is the best way to organize data transfer between server, frontend and vice versa.
We have `camelCase` model attributes on server that are being serialized to response as `snake_case`.

Then we have our frontend that works with this data and sends it in `snake_case` to the server, where validator catches it and validates in `camelCase`, because then in controller it is being merged right away to model, where we use the same `camelCase`.

The problem here is this communication between frontend and server, we send `snake_case`, because we got from the server in this format, isn't it better to automatically convert it to `camelCase` somewhere in validator, so it will understand `first_name` as a variant of `firstName`?

**Comment by McSneaky** [comment, marked as answer]
I just used https://www.npmjs.com/package/camelcase-keys for it :) 

Would be cool if it's configurable tho, but that would be super small (and maybe useless) feature / complexity to the core
